src/take.fypp

```diff
@@ -94,19 +107,130 @@ CONTAINS
         ! Note how +1 is needed here to account for 1-based Fortran indexing
         take => vec( (n+sigma+1): )
 
-      END IF
+      END IF ! +/- take
 
     ELSE
 
       ! Overtake
       NULLIFY(take)
 
-    END IF
+    END IF ! lvalid
 
+    RETURN
   END FUNCTION take_sv_${k1}$_${k2}$
+
+  #:endfor
+#:endfor
+#:for k1, t1 in INT_KINDS_TYPES
+  #:for k2, t2 in IR_KINDS_TYPES
+!===============================================================================
+! Implementation of scalar take vector for ${k1} scalar and ${k2} assumed-rank array
+!===============================================================================
+  FUNCTION take_sr_${k1}$_${k2}$ ( sigma, arr ) RESULT( take )
+    IMPLICIT NONE
+
+    ! Arguments
+    ${t1}$, INTENT(IN) :: sigma
+    ${t2}$, INTENT(IN), TARGET :: arr(..)
+    ${t2}$, POINTER :: take(..)
```

Comment:
@jvdp1 Can you suggest how to fix this line? GFortran does not like it. It produces the following error message:
```
Error: Assumed-rank array at (1) must be a dummy argument
```

Reply:
Mmmh... I don't know. It seems to be the same issue as with @arjenmarkus 's code.
I though gfortran supported assumed-rank arrays. Was I wrong?

Reply:
@jvdp1 Yeah, it says so here that assumed-rank is supported since GFortran 9.0, and `SELECT RANK` since 10.0: https://gcc.gnu.org/wiki/Fortran2018Status

Reply:
...and the bug report is marked invalid for violating the standard. I'll push a commit to get this to compile with GFortran 10.3.0.